simleague/simset.py

```diff
@@ -180,10 +180,35 @@ async def halftimebreak(self, ctx, time: int = 1):
     @simset.command()
     async def resultchannel(self, ctx, channel: discord.TextChannel):
         """Add a channel for automatic result posting."""
-        async with self.config.guild(ctx.guild).resultchannel() as result:
-            result.append(channel.id)
+        channels = await self.config.guild(ctx.guild).resultchannel()
+        for c in channels:
+            if c == channel.id:
+                await ctx.send("Results are already posted in this channel")
+                return
+
+        channels.append(channel.id)
+        await self.config.guild(ctx.guild).resultchannel.set(channels)
         await ctx.tick()
 
+    @simset.command()
+    async def resultchannels(self, ctx, option: str):
+        """Show or clear all result channels."""
+        if option == "clear":
+            await self.config.guild(ctx.guild).resultchannel.set([])
+            await ctx.tick()
+        elif option == "show":
+            async with self.config.guild(ctx.guild).resultchannel() as result:
+                a = []
+                for res in result:
+                    channel = ctx.guild.get_channel(res).name
```

Comment:
If the channel is deleted, this will return None and fail with an attrib error

Reply:
Added check to make sure channel exists before appending to the list